src/Interpreters/SyntaxAnalyzer.cpp

```diff
@@ -835,6 +836,9 @@ SyntaxAnalyzerResultPtr SyntaxAnalyzer::analyzeSelect(
     {
         optimizeIf(query, result.aliases, settings.optimize_if_chain_to_miltiif);
 
+        /// Removing arithmetic operations from functions
+        ArithmeticOperationsInAgrFuncVisitor().visit(query);
```

Comment:
need a setting here and test for correctness: same result in stateless test with and without setting